cli/command/image/build.go

```diff
@@ -477,3 +479,25 @@ func replaceDockerfileTarWrapper(ctx context.Context, inputTarStream io.ReadClos
 
 	return pipeReader
 }
+
+func createBuildArgs(cfg *configfile.ConfigFile, o opts.ListOpts) map[string]*string {
+	permitted := map[string]string{
+		"HTTP_PROXY":  cfg.HTTPProxy,
+		"HTTPS_PROXY": cfg.HTTPSProxy,
+		"NO_PROXY":    cfg.NoProxy,
+		"FTP_PROXY":   cfg.FTPProxy,
+	}
+	m := runconfigopts.ConvertKVStringsToMapWithNil(o.GetAll())
+	for k, v := range permitted {
+		if v == "" {
+			continue
+		}
+		if _, ok := m[k]; !ok {
+			o.Set(fmt.Sprintf("%s=%s", k, v))
```

Comment:
Perhaps some Golang expert can help here as I'm out of my depth.

If instead of `o.Set` I try to `m[k] = &v`, the value of `m[k]` outside of this function is always `""`.

This makes no sense to me as when I try to reproduce in the [Go Playground](https://play.golang.org/p/Ho3XnzJCTS) I can't.
There is probably some subtle interaction happening here that I don't quite understand.

I'd like to fix this before merge so I can avoid calling `ConvertKVStringsToMapWithNil` twice.